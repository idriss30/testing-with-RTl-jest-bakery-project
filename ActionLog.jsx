import React from "react";

export const ActionLog = ({ actions }) => {
  return (
    <>
      <h2>Action Log</h2>
      <ul>
        {actions.map(({ time, message, data }, i) => {
          const date = new Date(time).toUTCString();
          return (
            <li key={i}>
              {date} - {message} - {JSON.stringify(data)}
            </li>
          );
        })}
      </ul>
    </>
  );
};
