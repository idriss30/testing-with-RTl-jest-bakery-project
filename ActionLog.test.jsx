import React from "react";
import { render, screen } from "@testing-library/react";
import { ActionLog } from "./ActionLog.jsx";

test("testing action log functions", () => {
  const dayToMs = (days) => days * 24 * 60 * 60 * 10000;
  const actions = [
    {
      time: new Date(dayToMs(1)),
      message: "initial inventory loaded",
      data: { cheesecake: 2, danish: 100, croissant: 30 },
    },
    {
      time: new Date(dayToMs(2)),
      message: "added chocolate-croissant",
      data: { "chocolate-croissant": 30 },
    },
    {
      time: new Date(dayToMs(2)),
      message: "item removed",
      data: { croissant: 20 },
    },
  ];
  const { container } = render(<ActionLog actions={actions} />);
  expect(container).toMatchSnapshot();
});
