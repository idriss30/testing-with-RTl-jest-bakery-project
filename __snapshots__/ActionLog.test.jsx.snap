// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing action log functions 1`] = `
<div>
  <div
    data-testid="action-log"
  >
    <h2>
      Action Log
    </h2>
    <ul>
      <li>
        Sun, 11 Jan 1970 00:00:00 GMT
         - 
        initial inventory loaded
         - 
        {"cheesecake":2,"danish":100,"croissant":30}
      </li>
      <li>
        Wed, 21 Jan 1970 00:00:00 GMT
         - 
        added chocolate-croissant
         - 
        {"chocolate-croissant":30}
      </li>
      <li>
        Wed, 21 Jan 1970 00:00:00 GMT
         - 
        item removed
         - 
        {"croissant":20}
      </li>
    </ul>
  </div>
</div>
`;
